python-avd/pyavd/_eos_designs/structured_config/underlay/utils.py

```diff
@@ -33,20 +33,11 @@ def _avd_peers(self: AvdStructuredConfigUnderlayProtocol) -> list:
         """
         return natural_sort(get(self._hostvars, f"avd_topology_peers..{self.shared_utils.hostname}", separator="..", default=[]))
 
-    @cached_property
-    def _underlay_filter_peer_as_route_maps_asns(self: AvdStructuredConfigUnderlayProtocol) -> list:
-        """Filtered ASNs."""
-        if not self.inputs.underlay_filter_peer_as:
-            return []
-
-        # using set comprehension with `{}` to remove duplicates and then run natural_sort to convert to list.
-        return natural_sort({link["peer_bgp_as"] for link in self._underlay_links if link["type"] == "underlay_p2p"})
-
     @cached_property
     def _underlay_links(self: AvdStructuredConfigUnderlayProtocol) -> list:
         """Returns the list of underlay links for this device."""
         underlay_links = []
-        underlay_links.extend(self._uplinks)
+        underlay_links.extend(get(self._hostvars, "switch.uplinks"))
```

Comment:
@ClausHolbechArista we don't have a better solution for this today right?

Reply:
Please don't refactor stuff around facts. It will only make the facts PR harder to merge.